src/Core/Resolvers/MsSqlQueryBuilder.cs

```diff
@@ -62,6 +69,51 @@ public string Build(SqlQueryStructure structure)
             return query;
         }
 
+        /// <summary>
+        /// Build left and right predicate operand and resolve the predicate operator into
+        /// {OperandLeft} {Operator} {OperandRight}
+        /// </summary>
+        protected override string Build(Predicate? predicate)
+        {
+            if (predicate is null)
+            {
+                throw new ArgumentNullException(nameof(predicate));
+            }
+
+            StringBuilder predicateString = new();
+
+            if (predicate.Left is not null)
+            {
+                string leftOperand = ResolveOperand(predicate.Left);
+                string operation = Build(predicate.Op);
+                string rightOperand = ResolveOperand(predicate.Right);
+
+                // This is done to make sure there are performance issues due to implicit conversion of data types from nvarchar to varchar.
+                if (predicate.Left.AsColumn() is not null && predicate.Left.AsColumn()!.ColumnSqlDbType == SqlDbType.VarChar && !rightOperand.Equals("NULL", StringComparison.OrdinalIgnoreCase))
+                {
+                    // This operation is valid even when the column being filtered is not VARCHAR(MAX), but a smaller VARCHAR type such as VARCHAR(20).
+                    rightOperand = $"CAST({rightOperand} AS VARCHAR(MAX))";
+                }
+
+                // For Binary predicates:
```

Comment:
Why cant the right side be a Column? What if both sides are columns i.e. its a Join Predicate?

Reply:
removed casting approach, the new approach works for both sides